kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/exception/BigQueryStorageWriteApiErrorResponses.java

```diff
@@ -75,4 +99,29 @@ public static boolean hasInvalidSchema(Collection<String> messages) {
     public static boolean isStreamClosed(String errorMessage) {
         return errorMessage.contains(STREAM_CLOSED);
     }
+
+    /**
+     * Tells if the exception is of storage exception type and  error code belong to the list of non retriable
+     * storage error code.
+     *
+     * @param exception Exception received from Batch mode data ingestion
+     * @return Retruns true if the exception is non-retriable
+     */
+    public static boolean isNonRetriableStorageError(Exception exception) {
+        Exceptions.StorageException storageException = null;
+        Throwable t = exception.getCause();
+        if (t instanceof StatusRuntimeException || t instanceof StatusException) {
+            storageException = Exceptions.toStorageException(exception);
+        }
+        if (storageException == null) {
```

Comment:
nit: this can just be an else block.

Reply:
No. line 114 can result in null as well [ref](https://github.com/googleapis/java-bigquerystorage/blob/main/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/Exceptions.java#L144). Thus this needs to be in if block